Library/Homebrew/formula.rb

```diff
@@ -2428,6 +2428,14 @@ def uses_from_macos(dep)
       specs.each { |spec| spec.uses_from_macos(dep) }
     end
 
+    def on_macos(&_block)
+      yield
```

Comment:
What do you think about having the `yield` versions be in `extend/os`?

Reply:
Does this mean removing the methods from the main formula class? Or can I just define the method signature here?

Reply:
Define empty methods here.